sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ImplementISerializableCorrectly.CSharp9.cs

```diff
@@ -0,0 +1,282 @@
+        { /*do something*/ }
+        protected SerializableDerived_New_GetObjectData(SerializationInfo info, StreamingContext context)
+            : base(info, context)
+        { /*do something*/ }
+
+        public new void GetObjectData(SerializationInfo info, StreamingContext context)
+        //FN, should be second location with Make 'GetObjectData' 'public' and 'virtual', or seal 'SerializableDerived_New_GetObjectData'.}}
+        {
+            base.GetObjectData(info, context);
+        }
+    }
+
+    [Serializable]
+    public record SerializableDerived_Not_Overriding_GetObjectData : SerializableRecord
+    // FN, Should be Non-compliant
+    // FN, should be second location with @ -1 Override 'GetObjectData(SerializationInfo, StreamingContext)' and serialize 'serializableField'.}}
+    {
+        private SerializableRecord serializableField;
+        public SerializableDerived_Not_Overriding_GetObjectData()
+        { /*do something*/ }
+        protected SerializableDerived_Not_Overriding_GetObjectData(SerializationInfo info, StreamingContext context)
+            : base(info, context)
+        { /*do something*/ }
+    }
+
+    [Serializable]
+    public partial record Partial_SerializableDerived_CallingBase_GetObjectData_Record : SerializableRecord
+    {
+        public Partial_SerializableDerived_CallingBase_GetObjectData_Record() { }
+        protected Partial_SerializableDerived_CallingBase_GetObjectData_Record(SerializationInfo info, StreamingContext context) : base(info, context) { }
+        public override partial void GetObjectData(SerializationInfo info, StreamingContext context);
+    }
+
+    public partial record Partial_SerializableDerived_CallingBase_GetObjectData_Record
+    {
+        public override partial void GetObjectData(SerializationInfo info, StreamingContext context) =>
+            base.GetObjectData(info, context);
+    }
+
+    [Serializable]
+    public partial record Partial_SerializableDerived_Not_CallingBase_GetObjectData_Record : SerializableRecord
+    // FN, Should be Non-compliant
+    {
+        public Partial_SerializableDerived_Not_CallingBase_GetObjectData_Record() { }
+        protected Partial_SerializableDerived_Not_CallingBase_GetObjectData_Record(SerializationInfo info, StreamingContext context) : base(info, context) { }
+        public override partial void GetObjectData(SerializationInfo info, StreamingContext context);
+    }
+
+    public partial record Partial_SerializableDerived_Not_CallingBase_GetObjectData_Record
+    // FN, Should be Non-compliant
+    {
+        public override partial void GetObjectData(SerializationInfo info, StreamingContext context)
+        //FN, should be second location with Invoke 'base.GetObjectData(SerializationInfo, StreamingContext)' in this method.}}
+        { /*do something*/ }
+    }
+
+    [Serializable]
+    public partial class Partial_SerializableDerived_CallingBase_GetObjectData_Class : SerializableClass
```

Comment:
make this a record? same for `Partial_SerializableDerived_CallingBase_GetObjectData_Class` and `Partial_SerializableDerived_Not_CallingBase_GetObjectData_Class`?